CSharp/Library/Microsoft.Bot.Builder.Azure/Utils.cs

```diff
@@ -156,7 +156,7 @@ public static string GetStateApiUrl(string key = AppSettingKeys.StateEndpoint)
             return result;
         }
 
-        private static readonly Regex AzureFunctionAssembly = new Regex(@"(\S+)#\d+-\d+", RegexOptions.Compiled);
+        private static readonly Regex AzureFunctionAssembly = new Regex(@"(\S+)__+-*\d+", RegexOptions.Compiled);
```

Comment:
>__+-*\d+ [](start = 79, length = 8)

"at least one nonspace character, followed by at least two underscores, followed by any number of dashes, followed by at least one digit" - does it need to be so specific?

Reply:
the underscore is what divides the regex and allows us to extract the module name. If we do \S+\d+ the numbers and underscores fall under \S+ and we get the entire string which is wrong. I do see how specific it is, and potential issues with it. Right now, this is fully broken, so I'll merge this in which will guaranteed support more scenarios than our currently live version, and adjust once I get confirmation from functions team. Let me know if that doesn't seem reasonable :)